mongosql/src/ast/rewrites/aggregate.rs

```diff
@@ -165,6 +174,59 @@ impl Visitor for AggregateUsageCheckVisitor {
     }
 }
 
+#[derive(Default)]
+pub struct MultiArgCountVisitor {
```

Comment:
This was not in the design doc. The `algebrizer` rejects aggregations with multiple arguments, and the `mir` and `air` only support single-argument accumulator exprs. I figured the smallest change was to syntactically rewrite multi-column `COUNT`s into `COUNT(<doc>)`, thus making it an aggregation with a single argument. The previous ticket already enabled `COUNT` with document arguments so this change requires no further work down the line except desugarer updates.